sklearn/pipeline.py

```diff
@@ -481,7 +481,7 @@ def _inverse_transform(self, X):
         return Xt
 
     @if_delegate_has_method(delegate='_final_estimator')
-    def score(self, X, y=None):
+    def score(self, X, y=None, **score_params):
```

Comment:
The difference with `**fit_params` is that transformers can have `.fit` method, but there is only a single .score method - that's why I went with `**score_params` without `__` handling. But I see the problem now.

So did I get it right: add sample_weight=None argument, don't pass sample_weight to .score if it is None (as not all score methods suport sample_weight), and in the future scikit-learn can add support for `__`-delimeted keyword arguments which are passed to .transform methods?

`score(X, y=None, sample_weight=None)` signature has a small issue: it is not possible to distinguish between passing `sample_weight=None` and not passing `sample_weight`; it is technically different if estimator.score doesn't support sample_weight argument. Should I handle this case by making signature `.score(X, y=None, **kwargs)`?